module/zfs/dsl_dir.c

```diff
@@ -304,6 +387,10 @@ dsl_dir_hold_obj(dsl_pool_t *dp, uint64_t ddobj,
 			dd = winner;
 		} else {
 			spa_open_ref(dp->dp_spa, dd);
+
+			if (dd->dd_myname[0] != '$') {
```

Comment:
FYI, I considered excluding '%' here (temporary snapshots), and I think I _just_ come down on the side of leaving it out, ie, no change. They're not really seen by the user, but they _are_ real work, and so should be counted.